parser/prism/Translator.cc

```diff
@@ -284,7 +284,14 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
 
             auto expr = translate(blockArg->expression);
 
-            return make_unique<parser::BlockPass>(location, move(expr));
+            if (!hasExpr(expr)) {
```

Comment:
I've been seeing checks of `directlyDesugar` in addition to `hasExpr` on other branches, is that worth doing here?
```suggestion
            if (!directlyDesugar || !hasExpr(expr)) {
```